### PR TITLE
Google Login Page

### LINKED ISSUE #809: Google Login Page
Make a google-login page - with React-hook

Google auth2.0 and login with react tutorial: 
https://dev.to/sivaneshs/add-google-login-to-your-react-apps-in-10-mins-4del

NPM google-login package:
https://www.npmjs.com/package/react-google-login

### PR DESCRIPTION
**Pull request by Silenchezhian**
Created as the frontend for SJSU SCE discord bot -- allows users to login and logout with gmail account. 
image

If successful, user gets a message to type verify in discord channel and user information is printed in console.
image

If failed, user gets an alert that the login failed. 
image

Need to generate and add REACT_APP_GOOGLE_CLIENT_ID to config.json in src. 

Resolves #809

### CODE REVIEW

**Review by ThaiQ — CHANGES_REQUESTED** [review]
Looking good!
1 changes: Please remove `package.json` and `package-lock.json` from commit. Follow this [link](https://devconnected.com/how-to-remove-files-from-git-commit/). You do ***NOT*** need to `git rm --cached <file>`.

**Comment by ThaiQ** [comment]
also - for future references, name your branch as `firstname/topic` - ex: `sahana/googlelogin`

**Review by ThaiQ — APPROVED** [review]

**Review by charlotte-zhuang — COMMENTED** [review]
I guess we wouldn't want to merge this in to dev, since it's currently console logging the response from Google?

I'm also not sure where to get the react app google client id, so that might be causing some issues on my end.

1. I think we need to commit `package.json` at least, and preferably also a working `package-lock.json`
2. Nothing is logged to the console after I sign in with Google
3. There's no logout button

**Comment by ThaiQ** [comment]
We are not merging yet - work in progress

File: src/Pages/SJSUDiscordBot/GoogleLogin.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+import { ApiResponse } from '../../../src/APIFunctions/ApiResponses';
+const clientId = REACT_APP_GOOGLE_CLIENT_ID;
+
+function Verify(){
+  const [isLogined, setLoginState] = useState(false);
+  const [accessToken, setAccessToken] = useState('');
+  const [username, setUsername] = useState('');
+  const [id, setID] = useState(window.location.href.split('/')[5]);
+
+  const getTempUser = () => {
```
Comment (evanugarte) [inline src/Pages/SJSUDiscordBot/GoogleLogin.js:15]: can we put this in `APIFunctions`?

File: src/Pages/SJSUDiscordBot/GoogleLogin.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        <h4> You are verifying as <b>{username}</b></h4>
+        <h5> Please login with SJSU email</h5>
+        { isLogined ?
+          <GoogleLogout
+            clientId= {clientId}
+            buttonText='Logout'
+            onLogoutSuccess={ logout }
+            onFailure={ onFailure }
+          >
+          </GoogleLogout>: <GoogleLogin
```
Comment (evanugarte) [inline src/Pages/SJSUDiscordBot/GoogleLogin.js:63]:
```suggestion
          <GoogleLogout
            clientId= {clientId}
            buttonText='Logout'
            onLogoutSuccess={ logout }
            onFailure={ onFailure }
          />: <GoogleLogin
```

**Review by ThaiQ — APPROVED** [review]
Clean!

**Review by ThaiQ — APPROVED** [review]

File: src/Pages/SJSUDiscordBot/GoogleLogin.js [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+  const [discordID, setDiscordID] = useState(0);
+  const [id, setID] = useState(window.location.href.split('/')[5]);
+  const [validID, setValidID] = useState(false);
+  const [loginFailed, setLoginFailed] = useState(false);
+  const [logoutFailed, setLogoutFailed] = useState(false);
+  const [SJSUEmail, setSJSUEmail] = useState(false);
+
+  const getUser = async () => {
+    await getTempUser(id).then(res => {
+      if(!res.error){
```
Comment (charlotte-zhuang) [inline src/Pages/SJSUDiscordBot/GoogleLogin.js:20]:
```suggestion
      if (!res.error){
```
Minor complaint: it's better style to add a space between `if ()`

File: src/Pages/SJSUDiscordBot/GoogleLogin.js [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+  return(
+    <div>
+      <section style={{margin: '100px'}}>
+        <h1>SJSU SCE Discord Verify</h1>
+        { !validID?
+          <h5>Invalid ID: please type verify in the discord chat again</h5>
+          : isLogined ?
+            logoutFailed ?
+              <>
+                <h5>Logout failed. Please try again or contact us. </h5>
```
Comment (charlotte-zhuang) [inline src/Pages/SJSUDiscordBot/GoogleLogin.js:69]:
```suggestion
                <h6>Logout failed. Please try again or contact us. </h6>
```
Nested headings should be a higher level.

**Review by charlotte-zhuang — COMMENTED** [review]
This PR requires installing the `react-google-login` package.

I'm not sure how to test the functionality.

**Pull request closed without merging**